src/main/java/io/github/deweyjose/graphqlcodegen/Validations.java

```diff
@@ -0,0 +1,70 @@
+package io.github.deweyjose.graphqlcodegen;
+
+import java.io.File;
+import java.util.*;
+
+public class Validations {
+
+  /**
+   * We sort the input files by their paths to ensure a consistent order for processing.
+   *
+   * We maintain three sets:
+   *   validatedFiles to store the files that pass all checks,
+   *   encounteredDirectories to keep track of encountered directories
+   *   encounteredPaths to keep track of encountered paths.
+   *
+   * We iterate through the sorted files and perform the following checks:
+   *   Directory overlap: Skip files that are in directories already encountered.
+   *   Duplicate files: Skip files that are duplicates.
+   *   Ancestry in already added directories: Skip files that have ancestry in directories already encountered.
+   *
+   * Files that pass all checks are added to the validatedFiles set, and the corresponding
+   * directories and paths are updated in the encountered sets.
+   * @param files
+   */
+  public static void verifySchemaPaths(Collection<File> files) {
+    // Sort the input files to ensure consistent order
+    List<File> sortedFiles = new ArrayList<>(files);
+    sortedFiles.sort(Comparator.comparing(File::getPath));
+
+    Set<File> encounteredDirectories = new HashSet<>();
+    Set<String> encounteredPaths = new HashSet<>();
+
+    for (File file : sortedFiles) {
+      String path = file.getPath();
+
+      // Check for directory overlap
+      File parent = file.getParentFile();
+      if (parent != null && encounteredDirectories.contains(parent)) {
+        throw new IllegalArgumentException(
+          String.format("Overlap detected: %s contains %s", parent.getPath(), file.getPath())
```

Comment:
@mjfaga sorry I renamed this and lost your comment.

My first thought was to avoid any ambiguity and confusion around misconfig. If configured duplicate directories toss an error and fault, let folks fix the config.

But, looking at this again, its not even possible as `encounteredDirectories` is a set, so it naturally de-dupes. I'll remove this exception.